### PR TITLE
Penalty per load dimension

### PR DESCRIPTION
**Pull request by N-Wouda**
This PR generalises the load penalty system from a single penalty for all loads, to one per load dimension. Part of #458.

<details>

**Notes**:

Please read our [contributing guidelines](https://pyvrp.org/dev/contributing.html) first.
In particular:

- You must add tests when making code changes.
  This keeps the code coverage level up, and helps ensure the changes work as intended.
- When fixing a bug, you must add a test that would produce the bug in the master branch, and then show that it is fixed with the new code. 
- New code additions must be well formatted. Changes should pass the pre-commit workflow, which you can set up locally using [pre-commit](https://pre-commit.com/#intro). 
- Docstring additions must render correctly, including escapes and LaTeX.
- Finally, it is essential that all contributions in this PR are license-compatible with PyVRP's MIT license.
  Please check that this PR can be included into PyVRP under the MIT license.

</details>

### CODE REVIEW

**Comment by N-Wouda** [comment]
@wouterkool @leonlan could you review this?

File: tests/test_CostEvaluator.py [hunk 1]
```diff
@@ -17,24 +17,24 @@ def test_load_penalty():
...
 
-    assert_equal(cost_evaluator.load_penalty(0, 1), 0)  # below capacity
-    assert_equal(cost_evaluator.load_penalty(1, 1), 0)  # at capacity
+    assert_equal(cost_evaluator.load_penalty(0, 1, 0), 0)  # below capacity
+    assert_equal(cost_evaluator.load_penalty(1, 1, 0), 0)  # at capacity
 
     # Penalty per unit excess capacity is 2
-    # 1 unit above capacity
-    assert_equal(cost_evaluator.load_penalty(2, 1), 2)
+    # 1 unit above , 0capacity
```
Comment (leonlan) [inline tests/test_CostEvaluator.py:26]:
```suggestion
    # 1 unit above capacity
```

File: tests/test_CostEvaluator.py
Comment (leonlan) [inline tests/test_CostEvaluator.py:1]: Could we add a test that checks load penalty for non-zero dimension?

**Review by leonlan — APPROVED** [review]
LGTM

File: tests/test_CostEvaluator.py
Comment (leonlan) [reply tests/test_CostEvaluator.py:1]: Context: we currently only test `cost_eval.load_penalty(0, 0, dim)` for `dim = 0`.
